Fix Avalonia Obsolete Warnings

Reported via https://github.com/icsharpcode/AvaloniaILSpy/pull/82#issuecomment-973783612

@jeffreye we should consider fixing those soonish because who knows which Avalonia version will turn obsoletions into actual build breaks.

```
2>D:\GitWorkspace\AvaloniaILSpy\SharpTreeView\EditTextBox.cs(21,27,21,44): warning CS0672: Member 'EditTextBox.OnTemplateApplied(TemplateAppliedEventArgs)' overrides obsolete member 'TemplatedControl.OnTemplateApplied(TemplateAppliedEventArgs)'. Add the Obsolete attribute to 'EditTextBox.OnTemplateApplied(TemplateAppliedEventArgs)'.
2>D:\GitWorkspace\AvaloniaILSpy\SharpTreeView\EditTextBox.cs(34,8,34,17): warning CS0108: 'EditTextBox.SelectAll()' hides inherited member 'TextBox.SelectAll()'. Use the new keyword if hiding was intended.
2>D:\GitWorkspace\AvaloniaILSpy\SharpTreeView\SharpTreeView.cs(98,47,98,74): warning CS0108: 'SharpTreeView.IsTextSearchEnabledProperty' hides inherited member 'SelectingItemsControl.IsTextSearchEnabledProperty'. Use the new keyword if hiding was intended.
2>D:\GitWorkspace\AvaloniaILSpy\SharpTreeView\SharpTreeView.cs(101,15,101,34): warning CS0108: 'SharpTreeView.IsTextSearchEnabled' hides inherited member 'SelectingItemsControl.IsTextSearchEnabled'. Use the new keyword if hiding was intended.
2>D:\GitWorkspace\AvaloniaILSpy\SharpTreeView\SharpTreeView.cs(768,15,768,26): warning CS0108: 'SharpTreeView.UnselectAll()' hides inherited member 'ListBox.UnselectAll()'. Use the new keyword if hiding was intended.
2>D:\GitWorkspace\AvaloniaILSpy\SharpTreeView\EditTextBox.cs(23,4,23,29): warning CS0618: 'TemplatedControl.OnTemplateApplied(TemplateAppliedEventArgs)' is obsolete: 'Use OnApplyTemplate'
2>D:\GitWorkspace\AvaloniaILSpy\SharpTreeView\SharpTreeViewItem.cs(60,8,60,21): warning CS0618: 'PointerPressedEventArgs.MouseButton' is obsolete: 'Use PointerPressedEventArgs.GetCurrentPoint(this).Properties'
2>D:\GitWorkspace\AvaloniaILSpy\SharpTreeView\SharpTreeViewItem.cs(64,9,64,21): warning CS0618: 'PointerPressedEventArgs.ClickCount' is obsolete: 'Use DoubleTapped event or Gestures.DoubleRightTapped attached event'

3>------ Build started: Project: ILSpy.Core, Configuration: Debug Any CPU ------
3>D:\GitWorkspace\AvaloniaILSpy\ILSpy.Core\Controls\DockedPane.cs(38,33,38,50): warning CS0672: Member 'DockedPane.OnTemplateApplied(TemplateAppliedEventArgs)' overrides obsolete member 'TemplatedControl.OnTemplateApplied(TemplateAppliedEventArgs)'. Add the Obsolete attribute to 'DockedPane.OnTemplateApplied(TemplateAppliedEventArgs)'.
3>D:\GitWorkspace\AvaloniaILSpy\ILSpy.Core\TreeNodes\ResourceNodes\ResourceTreeNode.cs(78,79,78,113): warning CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call.
3>D:\GitWorkspace\AvaloniaILSpy\ILSpy.Core\Commands\SaveCodeContextMenuEntry.cs(39,13,39,47): warning CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call.
3>D:\GitWorkspace\AvaloniaILSpy\ILSpy.Core\TextView\DecompilerTextView.xaml.cs(796,69,796,116): warning CS0618: 'IPointerDevice.GetPosition(IVisual)' is obsolete: 'Use PointerEventArgs.GetPosition'
3>D:\GitWorkspace\AvaloniaILSpy\ILSpy.Core\Controls\DockedPane.cs(40,13,40,38): warning CS0618: 'TemplatedControl.OnTemplateApplied(TemplateAppliedEventArgs)' is obsolete: 'Use OnApplyTemplate'
3>D:\GitWorkspace\AvaloniaILSpy\ILSpy.Core\MainWindow.xaml.cs(437,33,437,56): warning CS0618: 'NativeMenuItemSeperator' is obsolete: 'This class exists to maintain backwards compatibility with existing code. Use NativeMenuItemSeparator instead'
```

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -5,8 +5,8 @@
     </PropertyGroup>
 
     <PropertyGroup>
-        <AvaloniaVersion>0.10.11</AvaloniaVersion>
-        <AvaloniaEditVersion>0.10.9</AvaloniaEditVersion>
+        <AvaloniaVersion>0.10.13</AvaloniaVersion>
+        <AvaloniaEditVersion>0.10.12.2</AvaloniaEditVersion>
     </PropertyGroup>
 
 </Project>

--- a/ILSpy.Core/Commands/SaveCodeContextMenuEntry.cs
+++ b/ILSpy.Core/Commands/SaveCodeContextMenuEntry.cs
@@ -36,7 +36,7 @@ namespace ICSharpCode.ILSpy.TextView
     {
         public void Execute(TextViewContext context)
         {
-            Execute(context.SelectedTreeNodes);
+            _ = Execute(context.SelectedTreeNodes);
         }
 
         public bool IsEnabled(TextViewContext context) => true;

--- a/ILSpy.Core/Controls/DockedPane.cs
+++ b/ILSpy.Core/Controls/DockedPane.cs
@@ -35,16 +35,16 @@ namespace ICSharpCode.ILSpy.Controls
 			set { SetValue(TitleProperty, value); }
 		}
 
-        protected override void OnTemplateApplied(TemplateAppliedEventArgs e)
+        protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
         {
-            base.OnTemplateApplied(e);
-            Button closeButton = (Button)e.NameScope.Find<Button>("PART_Close");
-            if (closeButton != null)
-            {
-                closeButton.Click += closeButton_Click;
-            }
+            base.OnApplyTemplate(e);
+			Button closeButton = (Button)e.NameScope.Find<Button>("PART_Close");
+			if (closeButton != null)
+			{
+				closeButton.Click += closeButton_Click;
+			}
 
-        }
+		}
 		
 		void closeButton_Click(object sender, RoutedEventArgs e)
 		{

--- a/ILSpy.Core/Controls/DockedPane.cs
+++ b/ILSpy.Core/Controls/DockedPane.cs
@@ -35,15 +35,14 @@ namespace ICSharpCode.ILSpy.Controls
 			set { SetValue(TitleProperty, value); }
 		}
 
-        protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
-        {
-            base.OnApplyTemplate(e);
+		protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
+		{
+			base.OnApplyTemplate(e);
 			Button closeButton = (Button)e.NameScope.Find<Button>("PART_Close");
 			if (closeButton != null)
 			{
 				closeButton.Click += closeButton_Click;
 			}
-
 		}
 		
 		void closeButton_Click(object sender, RoutedEventArgs e)

--- a/ILSpy.Core/ILSpy.Core.csproj
+++ b/ILSpy.Core/ILSpy.Core.csproj
@@ -29,10 +29,10 @@
   <ItemGroup>
     <PackageReference Include="Avalonia" Version="$(AvaloniaVersion)" />
     <PackageReference Include="Avalonia.Controls.DataGrid" Version="$(AvaloniaVersion)" />
-    <PackageReference Include="ICSharpCode.Decompiler" Version="7.1.0.6543" />
-    <PackageReference Include="Microsoft.DiaSymReader" Version="1.3.0" />
-    <PackageReference Include="Microsoft.DiaSymReader.Converter.Xml" Version="1.1.0-beta2-21110-01" />
-    <PackageReference Include="Microsoft.VisualStudio.Composition" Version="16.4.11" />
+    <PackageReference Include="ICSharpCode.Decompiler" Version="7.2.0.6844" />
+    <PackageReference Include="Microsoft.DiaSymReader" Version="1.4.0" />
+    <PackageReference Include="Microsoft.DiaSymReader.Converter.Xml" Version="1.1.0-beta2-22164-02" />
+    <PackageReference Include="Microsoft.VisualStudio.Composition" Version="17.1.20" />
     <PackageReference Include="NuGet.Client" Version="4.2.0" />
     <PackageReference Include="System.IO.Compression" Version="4.3.0" />
     <PackageReference Include="Avalonia.Markup.Xaml.Loader" Version="$(AvaloniaVersion)" />

--- a/ILSpy.Core/Languages/CSharpILMixedLanguage.cs
+++ b/ILSpy.Core/Languages/CSharpILMixedLanguage.cs
@@ -101,7 +101,7 @@ namespace ICSharpCode.ILSpy
 				}
 			}
 
-			protected override void WriteInstruction(ITextOutput output, MetadataReader metadata, MethodDefinitionHandle methodDefinition, ref BlobReader blob)
+			protected override void WriteInstruction(ITextOutput output, MetadataReader metadata, MethodDefinitionHandle methodDefinition, ref BlobReader blob, int methodRva)
 			{
 				int index = sequencePoints.BinarySearch(blob.Offset, seq => seq.Offset);
 				if (index >= 0) {
@@ -128,7 +128,7 @@ namespace ICSharpCode.ILSpy
 						highlightingOutput?.EndSpan();
 					}
 				}
-				base.WriteInstruction(output, metadata, methodDefinition, ref blob);
+				base.WriteInstruction(output, metadata, methodDefinition, ref blob, methodRva);
 			}
 
 			HighlightingColor gray = new HighlightingColor { Foreground = new SimpleHighlightingBrush(Colors.DarkGray) };

--- a/ILSpy.Core/MainWindow.xaml.cs
+++ b/ILSpy.Core/MainWindow.xaml.cs
@@ -434,7 +434,7 @@ namespace ICSharpCode.ILSpy
 				{
 					if (topLevelMenuItems.Count > 0)
 					{
-						topLevelMenuItems.Add(new NativeMenuItemSeperator());
+						topLevelMenuItems.Add(new NativeMenuItemSeparator());
 					}
 					foreach (var entry in category)
 					{

--- a/SharpTreeView/EditTextBox.cs
+++ b/SharpTreeView/EditTextBox.cs
@@ -18,23 +18,17 @@ namespace ICSharpCode.TreeView
 			get { return Item.Node; }
 		}
 
-		protected override void OnTemplateApplied(TemplateAppliedEventArgs e)
-		{
-			base.OnTemplateApplied(e);
+        protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
+        {
+            base.OnApplyTemplate(e);
 			Init();
 		}
 
-		void Init()
+        void Init()
 		{
 			Text = Node.LoadEditText();
 			Focus();
 			SelectAll();
-		}
-
-		void SelectAll()
-		{
-			SelectionStart = 0;
-			SelectionEnd = Text.Length;
 		}
 
 		protected override void OnKeyDown(KeyEventArgs e)

--- a/SharpTreeView/EditTextBox.cs
+++ b/SharpTreeView/EditTextBox.cs
@@ -18,13 +18,13 @@ namespace ICSharpCode.TreeView
 			get { return Item.Node; }
 		}
 
-        protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
-        {
-            base.OnApplyTemplate(e);
+		protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
+		{
+			base.OnApplyTemplate(e);
 			Init();
 		}
 
-        void Init()
+		void Init()
 		{
 			Text = Node.LoadEditText();
 			Focus();

--- a/SharpTreeView/SharpTreeView.cs
+++ b/SharpTreeView/SharpTreeView.cs
@@ -95,14 +95,6 @@ namespace ICSharpCode.TreeView
 			set { SetValue(IsTextSearchCaseSensitiveProperty, value); }
 		}
 
-		public static readonly StyledProperty<bool> IsTextSearchEnabledProperty =
-			AvaloniaProperty.Register<SharpTreeView, bool>(nameof(IsTextSearchEnabled), defaultValue: true);
-
-		public bool IsTextSearchEnabled {
-			get { return (bool)GetValue(IsTextSearchEnabledProperty); }
-			set { SetValue(IsTextSearchEnabledProperty, value); }
-		}
-
 		public static bool GetShowAlternation(AvaloniaObject obj)
 		{
 			return (bool)obj.GetValue(ShowAlternationProperty);
@@ -763,11 +755,6 @@ namespace ICSharpCode.TreeView
 			foreach (var item in nodes) {
 				this.SelectedItems.Add(item);
 			}
-		}
-
-		public void UnselectAll()
-		{
-			SelectedItems.Clear();
 		}
 	}
 }

--- a/SharpTreeView/SharpTreeViewItem.cs
+++ b/SharpTreeView/SharpTreeViewItem.cs
@@ -57,7 +57,7 @@ namespace ICSharpCode.TreeView
 				base.OnPointerPressed(e);
 			}
 
-			if (e.MouseButton == MouseButton.Left) {
+			if (e.GetCurrentPoint(this).Properties.IsLeftButtonPressed) {
 				startPoint = e.GetPosition(this);
 				e.Pointer.Capture(this);
 

